internal/ais/workflow.go

```diff
@@ -154,10 +168,10 @@ func (w *Workflow) SessionHandler(ctx temporalsdk_workflow.Context, params *Work
 	if parseResult.UpdatedAreldaMetadataRelPath != "" {
 		var fetchMetadataResult FetchActivityResult
 		e = temporalsdk_workflow.ExecuteActivity(
```

Comment:
It seems inefficient to download the metadata files from the AMSS when we have a local copy in the SIP. We aren't making any changes to the `UpdatedAreldaMetadata.xml` or `metadata.xml` files in preservation are we?  Maybe we can pass the path of the relevant metadata file from Enduro to this workflow?

Reply:
A different workflow and maybe a different worker, running in a different place than preprocessing/Enduro, so we don't have access to the SIP.

But you are right. As next steps I was planning to investigate the option of generating a preprocessing identifier that could be returned in the preprocessing child workflow result and added to the poststorage workflow params. That way we could temporary upload the `UpdatedAreldaMetadata.xml` or `metadata.xml` in MinIO from preprocessing and fetch it from this workflow using that identifier. It will prevent a second file extraction from the AIP and it also may remove the need of parsing the METS file. But that will be in different PRs, probably after the work you are doing to merge the metadata files together.